sympy/geometry/line.py

```diff
@@ -1070,6 +1070,40 @@ def contains(self, other):
             return Point.is_collinear(self.p1, self.p2, other.p1, other.p2)
         return False
 
+    def coordinate(self, other):
+        """
+        Finds the parameter ``t`` for which
+        ``other == t * self.p1 + (1 - t) self.p2`` is True.
+
+        Raises
+        ======
+
+        ValueError
+            When ``other`` is not contained in the line.
+
+            When ``other`` is not a ``Point``.
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Line
+        >>> p1, p2, p3 = Point(0, 1), Point(2, 3), Point(4, 5)
+        >>> l = Line(p1, p2)
+        >>> l.coordinate(p3)
+        2
+        """
+        if isinstance(other, Point):
+            other=Point(other)
+            if Point.is_collinear(self.p1, self.p2, other):
+                if self.p1.x != self.p2.x:
+                    ti = (other.x - self.p1.x)/(self.p2.x - self.p1.x)
+                else:
+                    ti = (other.y - self.p1.y)/(self.p2.y - self.p1.y)
+                if ti.is_number:
```

Comment:
Is this necessary?

Reply:
Its for the case if `self.p2.y ==self.p1.y:`, then the else statement will explode . We can remove this if statement and convert the else statement to `elif self.p2.y !=self.p1.y:`  This will serve the same purpose.

Reply:
> convert the else statement to elif self.p2.y !=self.p1.y:

I think this is the proper way. It will allow the use of symbolic values for coordinates.